src/pages/index.js

```diff
@@ -78,24 +126,123 @@ addcreateCardButton.addEventListener("click", () => {
   formValidators["add-button-form"].resetValidation();
 });
 
+avatarImage.addEventListener("click", () => {
+  avatarEditModal.open();
+});
+
 // EVENT HANDLERS
 
+function handleLikeCard(cardId) {
+  return api
+    .likeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card._updateLikeStatus(true);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error liking card: ${err}`);
+    });
+}
+
+function handleUnlikeCard(cardId) {
+  return api
+    .unlikeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card._updateLikeStatus(false);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error unliking card: ${err}`);
+    });
+}
+
+function handleDeleteButtonClick(cardId) {
+  deleteCardModal.open(cardId);
+}
+
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo({
+  profileSaveButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```